client.go

```diff
@@ -302,15 +302,25 @@ func (c *Client) CreateDataset(options *CreateDatasetOptions) (datasetState *Dat
 	tickChan := time.NewTicker(tickerWaitingForVolume).C
 
 	for {
-		if s, err := c.GetDatasetState(p.DatasetID); err == nil {
+		var strErrDel string
+		s, err := c.GetDatasetState(p.DatasetID)
+		if err == nil {
 			return s, nil
 		} else if err != errStateNotFound {
-			return nil, err
+			errDel := c.DeleteDataset(p.DatasetID)
```

Comment:
:question: I'm not sure about this. Previously an error returned by `GetDatasetState` would be ignored and the function would continue to loop until the timeout....potentially recovering from a temporary error in the dataset state request. E.g. flocker-control is being restarted.